tasks/main.yml

```diff
@@ -5,7 +5,7 @@
     echo -n "bastion-"
     shuf -n 3 {{ role_path }}/files/castles.txt | xargs | tr " " "-"
   when: heat_stack_name is undefined
-  failed_when: heat_stack_name_gen.stderr_lines
+  failed_when: heat_stack_name_gen.stderr_lines is defined
```

Comment:
looks like stderr_lines is only defined in Ansible 2.3. I tried the following modification successfully with both 2.2.1 and 2.3.0

```yaml
  failed_when: heat_stack_name_gen.stderr|default('')
```